### PR TITLE
Java: add documentation for the various packages

### PR DESCRIPTION
**Pull request by johnoliver**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: java/PACKAGES.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+## Connectors
+
+`semantickernel-connectors-ai-openai`
+: Provides a connector that can be used to interact with an OpenAI API.
+
+### Memory Connectors
+
+#### JDBC Memory Connectors
+
+Provides a memory connector that can be used to interact with a JDBC database.
```
Comment (d3r3kk) [inline java/PACKAGES.md:51]: Do we want to make a brief mention of the JDBC package, and how one might want to  use it to extend a JDBC-based connection that we don't provide out of the box?

File: java/PACKAGES.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+- `semantickernel-connectors-memory-sqlite`
+- `semantickernel-connectors-memory-postgresql`
+- `semantickernel-connectors-memory-mysql`
+
+`semantickernel-connectors-memory-azurecognitivesearch`
+: Provides a memory connector that allows using Azure Cognitive Search as a memory.
```
Comment (d3r3kk) [inline java/PACKAGES.md:57]:
```suggestion
: Provides a memory connector for using Azure Cognitive Search as the memory provider for an application.
```

File: java/PACKAGES.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+### Miscellaneous Packages
+
+`semantickernel-planners`
+: Implementations of various planners that can be used to plan the execution of semantic functions.
```
Comment (d3r3kk) [inline java/PACKAGES.md:62]:
```suggestion
: Implementations of various planners that can be used in the execution of semantic functions.
```

File: java/PACKAGES.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+`semantickernel-plugin-core`
+: Several example plugins, many of which are used in the samples.
```
Comment (d3r3kk) [inline java/PACKAGES.md:65]:
```suggestion
: Several example plugins, many of which are used in the [Java Semantic Kernel samples](/java-development/java/samples).
```

**Review by d3r3kk — APPROVED** [review]
Wrote a bunch of stuff that may have value, but I'm not going to be put out if you don't agree with my flowery prose 😁.

File: java/PACKAGES.md (hunk 1 above)
Comment (brunoborges) [reply java/PACKAGES.md:51]: Yes, eventually we will need detailed documentation for each concept, similarly to how Spring Project is documented.

Example of Spring Data JDBC https://docs.spring.io/spring-data/jdbc/docs/current/reference/html/

File: java/PACKAGES.md (hunk 4 above)
Comment (brunoborges) [reply java/PACKAGES.md:65]: "Semantic Kernel for Java" is what we use at all times.

**Review by d3r3kk — APPROVED** [review]

**Review by milderhc — APPROVED** [review]

**Review by brunoborges — APPROVED** [review]
This is a good start!

**Pull request merged**